test_conformance/api/test_api_min_max.cpp

```diff
@@ -1680,31 +1679,33 @@ int test_min_max_local_mem_size(cl_device_id deviceID, cl_context context, cl_co
     error = clGetDeviceInfo( deviceID, CL_DEVICE_LOCAL_MEM_SIZE, sizeof( maxSize ), &maxSize, 0 );
     test_error( error, "Unable to get max local buffer size" );
 
-    // Device version should fit the regex "OpenCL [0-9]+\.[0-9]+ *.*"
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_VERSION, sizeof( buffer ), buffer, &length );
-    test_error( error, "Unable to get device version string" );
+    try
+    {
+        device_version = get_device_cl_version(deviceID);
+    } catch (const std::runtime_error &e)
```

Comment:
Fine for this change but longer-term we probably want to introduce a type for these exceptions and catch them in the framework code. Catching individual exceptions like this defeats the point...